project.js

```diff
@@ -368,9 +368,21 @@ function getSomeProjects(listOffset) {
-        project_filename = ddd.name.replace(/[^a-z0-9]/gi, '_').toLowerCase();
+        var project_filename;
+        var projXMLcode;
+
+        // Catch a project that contains a null project name
+        if (ddd && ddd.name) {
+            project_filename = ddd.name.replace(/[^a-z0-9]/gi, '_').toLowerCase();
+            projXMLcode = ddd.code.substring(42, ddd.code.length);
+        }
+        else {
+            project_filename = " ";
+            // allow the string to be pruned below and still have a character remaining.
+            // This will cause the code length test to fail and treat this as an empty project.
+            projXMLcode = "1234567";
```

Comment:
This does get the job done but it is fragile because it depends on  a related test that could change over time.